// react-in-viewport/dist/next/lib/handleViewport.js
var _excluded = ["onEnterViewport", "onLeaveViewport"];

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import { useRef, forwardRef } from 'react';
import hoistNonReactStatic from 'hoist-non-react-statics';
import useInViewport from './useInViewport';
import { noop, defaultOptions, defaultConfig } from './constants';
import { jsx as _jsx } from "react/jsx-runtime";

var isFunctionalComponent = function isFunctionalComponent(Component) {
  return typeof Component === 'function' && !(Component.prototype && Component.prototype.render);
};

var isReactComponent = function isReactComponent(Component) {
  return Component.prototype && Component.prototype.isReactComponent;
};

function handleViewport(TargetComponent, options, config) {
  if (options === void 0) {
    options = defaultOptions;
  }

  if (config === void 0) {
    config = defaultConfig;
  }

  var ForwardedRefComponent = /*#__PURE__*/forwardRef(function (props, ref) {
    var refProps = _extends({
      forwardedRef: ref
    }, isReactComponent(TargetComponent) && !isFunctionalComponent(TargetComponent) ? {
      ref: ref
    } : {});

    return /*#__PURE__*/_jsx(TargetComponent, _extends({}, props, refProps));
  });

  function InViewport(_ref) {
    var _ref$onEnterViewport = _ref.onEnterViewport,
        onEnterViewport = _ref$onEnterViewport === void 0 ? noop : _ref$onEnterViewport,
        _ref$onLeaveViewport = _ref.onLeaveViewport,
        onLeaveViewport = _ref$onLeaveViewport === void 0 ? noop : _ref$onLeaveViewport,
        restProps = _objectWithoutPropertiesLoose(_ref, _excluded);

    var node = useRef();

    var _useInViewport = useInViewport(node, options, config, {
      onEnterViewport: onEnterViewport,
      onLeaveViewport: onLeaveViewport
    }),
        inViewport = _useInViewport.inViewport,
        enterCount = _useInViewport.enterCount,
        leaveCount = _useInViewport.leaveCount;

    var injectedProps = {
      inViewport: inViewport,
      enterCount: enterCount,
      leaveCount: leaveCount
    };
    return /*#__PURE__*/_jsx(ForwardedRefComponent, _extends({}, restProps, injectedProps, {
      ref: node
    }));
  }

  var name = TargetComponent.displayName || TargetComponent.name || 'Component';
  InViewport.displayName = "handleViewport(" + name + ")";
  return hoistNonReactStatic(InViewport, ForwardedRefComponent);
}

export default handleViewport;
