// react-in-viewport/dist/umd/lib/handleViewport.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react", "hoist-non-react-statics", "./useInViewport", "./constants", "react/jsx-runtime"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react"), require("hoist-non-react-statics"), require("./useInViewport"), require("./constants"), require("react/jsx-runtime"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.react, global.hoistNonReactStatics, global.useInViewport, global.constants, global.jsxRuntime);
    global.handleViewport = mod.exports;
  }
})(typeof globalThis !== "undefined" ? globalThis : typeof self !== "undefined" ? self : this, function (_exports, _react, _hoistNonReactStatics, _useInViewport2, _constants, _jsxRuntime) {
  "use strict";

  _exports.__esModule = true;
  _exports["default"] = void 0;
  _hoistNonReactStatics = _interopRequireDefault(_hoistNonReactStatics);
  _useInViewport2 = _interopRequireDefault(_useInViewport2);
  var _excluded = ["onEnterViewport", "onLeaveViewport"];

  function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

  function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

  function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

  var isFunctionalComponent = function isFunctionalComponent(Component) {
    return typeof Component === 'function' && !(Component.prototype && Component.prototype.render);
  };

  var isReactComponent = function isReactComponent(Component) {
    return Component.prototype && Component.prototype.isReactComponent;
  };

  function handleViewport(TargetComponent, options, config) {
    if (options === void 0) {
      options = _constants.defaultOptions;
    }

    if (config === void 0) {
      config = _constants.defaultConfig;
    }

    var ForwardedRefComponent = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
      var refProps = _extends({
        forwardedRef: ref
      }, isReactComponent(TargetComponent) && !isFunctionalComponent(TargetComponent) ? {
        ref: ref
      } : {});

      return /*#__PURE__*/(0, _jsxRuntime.jsx)(TargetComponent, _extends({}, props, refProps));
    });

    function InViewport(_ref) {
      var _ref$onEnterViewport = _ref.onEnterViewport,
          onEnterViewport = _ref$onEnterViewport === void 0 ? _constants.noop : _ref$onEnterViewport,
          _ref$onLeaveViewport = _ref.onLeaveViewport,
          onLeaveViewport = _ref$onLeaveViewport === void 0 ? _constants.noop : _ref$onLeaveViewport,
          restProps = _objectWithoutPropertiesLoose(_ref, _excluded);

      var node = (0, _react.useRef)();

      var _useInViewport = (0, _useInViewport2["default"])(node, options, config, {
        onEnterViewport: onEnterViewport,
        onLeaveViewport: onLeaveViewport
      }),
          inViewport = _useInViewport.inViewport,
          enterCount = _useInViewport.enterCount,
          leaveCount = _useInViewport.leaveCount;

      var injectedProps = {
        inViewport: inViewport,
        enterCount: enterCount,
        leaveCount: leaveCount
      };
      return /*#__PURE__*/(0, _jsxRuntime.jsx)(ForwardedRefComponent, _extends({}, restProps, injectedProps, {
        ref: node
      }));
    }

    var name = TargetComponent.displayName || TargetComponent.name || 'Component';
    InViewport.displayName = "handleViewport(" + name + ")";
    return (0, _hoistNonReactStatics["default"])(InViewport, ForwardedRefComponent);
  }

  var _default = handleViewport;
  _exports["default"] = _default;
});
